test/core/end2end/cq_verifier.c

```diff
@@ -71,17 +70,16 @@ typedef struct expectation {
 struct cq_verifier {
   /* bound completion queue */
   grpc_completion_queue *cq;
-  /* the root/sentinal expectation */
-  expectation expect;
+  /* expectation list */
+  expectation *first_expectation;
+  expectation *last_expectation;
```

Comment:
Thanks, that makes sense (I didn't see the `add()` function carefully enough :))